modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockService.scala

```diff
@@ -67,7 +66,7 @@ object BlockService {
         val all = sources.combine(destinations)
         val addressBalanceChanges = all.map {
           case (address, txs) =>
-            val change = txs.foldLeft(BigInt(0L))(
+            val change = txs.foldLeft(0L)(
               (change, tx) =>
                 tx match {
                   case Transaction(`address`, _, amount, fee, _, _) => change - amount.coerce - fee.coerce
```

Comment:
If we are worried about overflowing `Long` it could theoretically happen here during `change` calculation as well.

Reply:
fixed